r/R/io.R

```diff
@@ -292,7 +292,7 @@ make_readable_file <- function(file, mmap = TRUE, compression = NULL, filesystem
   file
 }
 
-make_output_stream <- function(x, filesystem = NULL) {
+make_output_stream <- function(x, filesystem = NULL, compression = NULL) {
```

Comment:
One thing to watch out here: sometimes people name their parquet files `something.parquet.snappy`, but you wouldn't use a CompressedOutputStream for that, you'd pass the compression option to the parquet writer itself. I would guess that the make_readable_file() path handles this already, maybe that can be a model (or maybe it doesn't and needs to).

Reply:
So for the `parquet.snappy` or even `snappy.parquet` I think it works because "snappy" isn't included here: 
https://github.com/apache/arrow/blob/3df2e0568240d6b629c0a3163df21a1a2a160810/r/R/io.R#L325-L330

But if someone tried something like this we do get an error that isn't super informative. I _think_ this is outside this PR so could the resolution here be to open another ticket for this specifically? 
``` r
library(arrow, warn.conflicts = FALSE)
tf <- tempfile(fileext = ".parquet.gz")
write_parquet(data.frame(x = 1:5), tf, compression = "gzip", compression_level = 5)
read_parquet(tf)
#> Error: file must be a "RandomAccessFile"
```

Reply:
Sure, this fails on master too so ok to make a separate JIRA.